### PR TITLE
Fix CMakeLists #1015 Windows OpenMP runtime

### LINKED ISSUE #1015: Issue with OPENMP_RUNTIME set to "COMP"
If I set `OPENMP_RUNTIME` to `COMP`, cmake will add `/openmp` flag to msvc, but later on it will be test
`OpenMP_CXX_FOUND` variable causing the build to failed with error `OpenMP not found`.

CMakeLists.txt reference:
```cmake
if(NOT OPENMP_RUNTIME STREQUAL "NONE")
  if(WIN32)
    add_compile_options("/openmp")
  else()
    find_package(OpenMP)
    if(OpenMP_CXX_FOUND)
      add_compile_options(${OpenMP_CXX_FLAGS})
    endif()
  endif()

  if(OPENMP_RUNTIME STREQUAL "INTEL")
    # Find Intel libraries.
    find_library(IOMP5_LIBRARY iomp5 libiomp5md PATHS
      ${INTEL_ROOT}/lib
      ${INTEL_ROOT}/lib/intel64
      ${INTEL_ROOT}/compiler/lib/intel64
      ${INTEL_ROOT}/oneAPI/compiler/latest/windows/compiler/lib/intel64_win
      ${INTEL_ROOT}/oneapi/compiler/latest/linux/compiler/lib/intel64_lin
      ${INTEL_ROOT}/oneapi/compiler/latest/mac/compiler/lib
      )
    if(IOMP5_LIBRARY)
      list(APPEND LIBRARIES ${IOMP5_LIBRARY})
      message(STATUS "Using OpenMP: ${IOMP5_LIBRARY}")
    else()
      message(FATAL_ERROR "Intel OpenMP runtime libiomp5 not found")
    endif()
    if(WIN32)
      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /nodefaultlib:vcomp")
      set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /nodefaultlib:vcomp")
    endif()
  elseif(OPENMP_RUNTIME STREQUAL "COMP")
    if(OpenMP_CXX_FOUND)
      list(APPEND LIBRARIES ${OpenMP_CXX_LIBRARIES})
      message(STATUS "Using OpenMP: ${OpenMP_CXX_LIBRARIES}")
    else()
      message(FATAL_ERROR "OpenMP not found")
    endif()
  else()
    message(FATAL_ERROR "Invalid OpenMP runtime ${OPENMP_RUNTIME}")
  endif()
endif()
```

### PR DESCRIPTION
**Pull request by A2va**
I think that will do it like this, right ?

Fixes #1015.

### CODE REVIEW

**Review by guillaumekln — COMMENTED** [review]
I don't think this works. The next `else` will be executed where there is another `FATAL_ERROR`.

Maybe we can't just prevent the original error on Windows? Something like this:

```cmake
  elseif(OPENMP_RUNTIME STREQUAL "COMP")
    if(OpenMP_CXX_FOUND)
      list(APPEND LIBRARIES ${OpenMP_CXX_LIBRARIES})
      message(STATUS "Using OpenMP: ${OpenMP_CXX_LIBRARIES}")
    elseif(NOT WIN32)
      message(FATAL_ERROR "OpenMP not found")
    endif()
  else()
```

Note the `elseif(NOT WIN32)` to prevent the error from being raised.

**Comment by A2va** [comment]
I finally made it like this: 
```cmake
elseif(OPENMP_RUNTIME STREQUAL "COMP")
    if(WIN32) 
      add_compile_options("/openmp")
    else()
        find_package(OpenMP)
        if(OpenMP_CXX_FOUND)
          add_compile_options(${OpenMP_CXX_FLAGS})
          list(APPEND LIBRARIES ${OpenMP_CXX_LIBRARIES})
          message(STATUS "Using OpenMP: ${OpenMP_CXX_LIBRARIES}")
        else()
          message(FATAL_ERROR "OpenMP not found")
        endif()
    endif()
```
I'm not sure if `openmp` flag is needed by INTEL openmp library. Is there any particular reason why it was placed at the beginning of the scope?

**Comment by guillaumekln** [comment]
> I'm not sure if openmp flag is needed by INTEL openmp library. Is there any particular reason why it was placed at the beginning of the scope?

The OpenMP flag is actually needed when using the Intel OpenMP runtime. So you can't move the block like this.

File: CMakeLists.txt [hunk 1]
```diff
@@ -235,11 +235,11 @@ if(NOT OPENMP_RUNTIME STREQUAL "NONE")
       set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /nodefaultlib:vcomp")
       set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /nodefaultlib:vcomp")
     endif()
-  elseif(OPENMP_RUNTIME STREQUAL "COMP")
+  elseif(OPENMP_RUNTIME STREQUAL "COMP" and NOT WIN32)
```
Comment (guillaumekln) [inline CMakeLists.txt:238]:
```suggestion
  elseif(OPENMP_RUNTIME STREQUAL "COMP")
```

**Review by guillaumekln — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -235,11 +235,11 @@ if(NOT OPENMP_RUNTIME STREQUAL "NONE")
       set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /nodefaultlib:vcomp")
       set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /nodefaultlib:vcomp")
     endif()
-  elseif(OPENMP_RUNTIME STREQUAL "COMP")
+  elseif(OPENMP_RUNTIME STREQUAL "COMP)
```
Comment (guillaumekln) [inline CMakeLists.txt:238]:
```suggestion
  elseif(OPENMP_RUNTIME STREQUAL "COMP")
```

Almost there. You accidentally removed a quote.

I will approve the change after this fix.

**Review by guillaumekln — CHANGES_REQUESTED** [review]

**Pull request merged**